openssl-dynamic/src/main/c/ssl.c

```diff
@@ -889,21 +889,57 @@ TCN_IMPLEMENT_CALL(jint, SSL, getLastErrorNumber)(TCN_STDARGS) {
 }
 
 static void ssl_info_callback(const SSL *ssl, int where, int ret) {
-    int *handshakeCount = NULL;
+    tcn_ssl_state_t* state = NULL;
     if (0 != (where & SSL_CB_HANDSHAKE_START)) {
-        handshakeCount = (int*) tcn_SSL_get_app_data3((SSL*) ssl);
-        if (handshakeCount != NULL) {
-            ++(*handshakeCount);
+        if ((state = tcn_SSL_get_app_state(ssl)) != NULL) {
+            state->handshakeCount++;
         }
     }
 }
 
+static tcn_ssl_state_t* new_ssl_state(tcn_ssl_ctxt_t* ctx) {
+    if (ctx == NULL) {
+        return NULL;
+    }
+
+    tcn_ssl_state_t* state = OPENSSL_malloc(sizeof(tcn_ssl_state_t));
+    if (state == NULL) {
+        return NULL;
+    }
+    memset(state, 0, sizeof(tcn_ssl_state_t));
```

Comment:
Can you use `OPENSSL_zalloc` instead?

Reply:
as far as I know this was just added in 1.1.0 so to not add more ifdefs we should just memset by ourselves.